backend/src/db/providers/skillProvider.ts

```diff
@@ -1,23 +1,102 @@
-import { ObjectId } from 'mongoose';
+import mongoose, { ObjectId } from 'mongoose';
 
 import UserSkillModel from 'db/models/UserSkill';
-import { IUserSkill } from 'interfaces/Ientities/IUserSkill';
+import UserModel from 'db/models/User';
+import { IUserSkill, IUserSkillPopulated } from 'interfaces/Ientities/IUserSkill';
+import { ITechnologyGroup } from 'interfaces/Ientities/Iusers';
+import NotFoundError from 'classes/errors/clientErrors/NotFoundError';
+import { isEqualObjectId } from 'utils/comparator/ObjectId/compareObjectIds';
 
 const getUserSkills = async (userId: string): Promise<IUserSkill[]> => {
-  const skills: IUserSkill[] = await UserSkillModel.find({ user: userId }).lean();
+  const skills: IUserSkill[] = await UserSkillModel.find({ user: userId })
+    .select('-_id skill score')
+    .lean();
   return skills;
 };
 
-const addUserSkill = async (userId: string, skillId?: ObjectId) => {
-  await UserSkillModel.create({
-    user: userId,
+const attachSkillToUserProfile = async (userId: string, userSkillId: ObjectId) => {
```

Comment:
move all logic out of provider, follow plz single responsibility priciple

Reply:
for example we already have provider that return user info, we need to use it cause it was the point of creation single provider, use them from controller, then move all helpers method into helpers
if attachSkillToUser means update user data create method update and unique provider for that

Reply:
Agree with @SaXaPhonist